### PR TITLE
Mntor 806/update existing emails

### PR DESCRIPTION
**Pull request by toufali**
- update all existing emails with new style and content
- refactor email preview page to work with new email template
- display actual production html and css for mockups (instead of a duped, loose representation)
- allow test-sending to custom recipient/email client with mock data
- put the email preview route behind auth
- remove obsolete code relating to pre-fxa email and unused email content

### CODE REVIEW

**Comment by jwhitlock** [comment]
@toufali do you plan to fix the tests, or do you want a PR for your PR?

**Comment by toufali** [comment]
@jwhitlock 😆 was planning on fixing today.

**Review by jwhitlock — CHANGES_REQUESTED** [review]
Thanks @toufali! I think the new page looks pretty good, and I was able to send each type of email to myself.

Lots of questions, nits, and opportunities to cleanup. I think the required actions are:

* Think about injection attacks on `breachedEmail` in the regular code, and if we need to escape any HTML
* Remove `console.log`, or comment that we mean to leave it in.

File: db/DB.js
Comment (jwhitlock) [reply db/DB.js:5]: I'm working on dependency updates, which could cause clashes in `package.json`. Removing `knex-paginate` can wait until the rest of the PR is merged, if you are worried about rebases.

File: locales/en/email-strings.ftl
Comment (toufali) [reply locales/en/email-strings.ftl:111]: I agree – it made me grimace a bit.  I should have spoken up with content team but I didn't want to be too nit-picky.  I think we should create a ticket with your research and have updated copy translated for a future release.

File: views/partials/email_partials/report.hbs
Comment (toufali) [reply views/partials/email_partials/report.hbs:7]: Can we chat a bit more about this?

File: views/partials/email_partials/email-header.hbs
Comment (toufali) [reply views/partials/email_partials/email-header.hbs:18]: great question -- what does Relay do?  Translating alt text seems like the right thing to do, but increases translation complexity/LOE

File: views/partials/email_partials/alert.hbs
Comment (toufali) [reply views/partials/email_partials/alert.hbs:45]: I prefer the shorthand for brevity – especially when dealing with inline styles.  This one probably just got missed, thanks for flagging.

File: locales/en/email-strings.ftl
Comment (toufali) [reply locales/en/email-strings.ftl:116]: I think it's worth creating a ticket for content (Chris) with this and your other copy suggestions.  I'm sensitive about changing "locked" copy at this stage, but happy to get it updated on a second round!

File: controllers/email-l10n.js
Comment (toufali) [reply controllers/email-l10n.js:26]: Hmm... I _believe_ previously the page was accessible to all on prod (not admin-protected for anyone).  IIRC, we even had a secops bug filed for a vuln with the partial params.  (oh yeah, we did, and I added this: https://github.com/mozilla/blurts-server/commit/3d4793ffebe0d38da3547f9b36edecb3697c8b82)

These changes put the page further on lockdown with admin-only access (in particular because we added email send functionality).  Totally agreed everyone should get a 404 on prod in a perfect world.  Currently accepting non-complex suggestions if it's worth the effort! 😅

File: views/layouts/email_l10n_mockups.hbs
Comment (toufali) [reply views/layouts/email_l10n_mockups.hbs:79]: hmm... I like the context that the placeholder provides, and – in many cases the user (QA, other FEs) will want to test other email clients (Outlook, Yahoo, etc) to test how the email looks.  I'd personally prefer not pre-filled.

File: views/partials/email_partials/email-footer.hbs
Comment (toufali) [reply views/partials/email_partials/email-footer.hbs:14]: oooh!  Great catch!

File: views/partials/email_partials/email-footer.hbs
Comment (toufali) [reply views/partials/email_partials/email-footer.hbs]: Updated this and previous comment via 89725febc33d5b9aec55dc126e4ed9ee84fbf660
Confirmed working on mockup emails, but untested on live emails – Could use a sanity check.

File: views/partials/email_partials/email-header.hbs
Comment (toufali) [reply views/partials/email_partials/email-header.hbs:16]: Assuming you mean this is broken on test emails sent via local.  It should work for preview page.  If you send to the Mailinator account the images show up, I believe (tho I don't exactly know why – I think it's because Gmail replaces images and Mailinator does not.) I confirmed it's working via Heroku.

File: template-helpers/emails.js
Comment (toufali) [reply template-helpers/emails.js:102]: yeah.  I remember being bummed when I wrote this.  Different email types require different footers...

File: controllers/email-l10n.js
Comment (toufali) [reply controllers/email-l10n.js:121]: Correct this is stray.

File: controllers/email-l10n.js
Comment (toufali) [reply controllers/email-l10n.js]: yeah... I believe this is just for the mockup though.  The link ends up being `http://localhost:6060/?utm_source=fx-monitor&utm_medium=email&utm_campaign=email_verify&utm_content=dashboard-cta`, which I figured was fine since we don't _really_ want to verify (or unsubscribe) a user.

Not sure how you're getting an undefined param in your link?

File: views/partials/email_partials/email-header.hbs
Comment (Vinnl) [reply views/partials/email_partials/email-header.hbs:18]: We translate alt text.

Comment (Vinnl) [reply views/partials/email_partials/email-header.hbs:18]: Although in this case, I'd leave the alt attribute empty - "Person at computer" doesn't really provide any meaningful info. Generally, I'd only add alt text (and localise alt text) if there's information in the image itself, which should also be localised if possible.

Comment (toufali) [reply views/partials/email_partials/email-header.hbs:18]: Thanks @Vinnl – I'll set it to empty string 👍

File: controllers/email-l10n.js
Comment (jwhitlock) [reply controllers/email-l10n.js:26]: I was going to suggest omitting the routes in prod, so everyone gets a 404, but it looks like we already do that:

https://github.com/mozilla/blurts-server/blob/cd4e437e6e1d13fe752a2c12474fb33f861e9a66/server.js#L243

So, I'm wrong, everyone gets a 404 in prod!

File: controllers/email-l10n.js
Comment (jwhitlock) [reply controllers/email-l10n.js]: I'm OK leaving a dashboard link for the mockup. I think that sets my expectations for these pages - layout is correct, but links and `utm_campaign` can be subtly different.

It was two weeks ago, but I believe it was from manually calling `EmailUtils.getVerificationUrl()` without a request user. In the test monthly breach email, I do get a valid token in the `Unsubscribe` link.

File: views/partials/email_partials/report.hbs
Comment (jwhitlock) [reply views/partials/email_partials/report.hbs:7]: We had a chat about this. The potential issue is not the demo email page, which is locked down to admins. Instead, I'm concerned about when this same report is sent to end users using an email that they entered into the site, since we don't do any HTML escaping of the `breachedEmail`.

We have some protections, such as email validation, to prevent invalid data getting into the database in the first place, so we have quite a few defenses. Additionally, an attacker would need to construct a string that is also a valid email, or the email won't be delivered - I wasn't able to find an example with a quick search.

Handlebars allows for [subexpressions](https://handlebarsjs.com/guide/expressions.html#subexpressions), and has an [escapeExpression](https://handlebarsjs.com/api-reference/utilities.html#handlebars-escapeexpression-string) helper, so this _might_ work:

```suggestion
      {{{ getString "email-breach-detected" email-address=( concat "<strong>" ( escapeExpression breachedEmail ) "</strong>" ) }}}
```

@toufali says that our implementation of handlebars doesn't include the full toolbox by default, and it may be a pain to add it, but he'll investigate.

**Review by jwhitlock — APPROVED** [review]
Thanks @toufali, I think this is ready to merge. Follow-on actions:

* HTML-escaping of emails (if it isn't done in this PR)
* File content suggestions as content tasks (I'll ask @chrisrooneys where he'd like them)

File: views/partials/email_partials/email-footer.hbs
Comment (jwhitlock) [reply views/partials/email_partials/email-footer.hbs]: Re-sent the test emails, footer links are good.

**Comment by toufali** [comment]
@jwhitlock added HTML-escape code -- you can see what I mean about getting access to the full HBS source.  If you can think of an easier way, I'm open (could probably just escape without HBS at this point).

Also, looking forward to simply getting rid of the "simply" on a future release 😅

**Pull request merged**
